src/integrationTest/groovy/wooga/gradle/unity/UnityIntegrationSpec.groovy

```diff
@@ -221,15 +221,26 @@ abstract class UnityIntegrationSpec extends IntegrationSpec {
         addTask(name, typeName, force, lines)
     }
 
+    // TODO: Update/add to gradle-commons-test
     void addTask(String name, String typeName, Boolean force, String... lines) {
         lines = lines ?: []
         buildFile << """
-        task (${name}, type: ${typeName}) {                       
+        def ${name} = tasks.register(\"${name}\"${typeName != null ? ", ${typeName}": ""}) {                       
```

Comment:
I less think of the performance gains but rather of consistently using the task provider API.

Reply:
But that doesn't make any difference in the test code block. Because we never really generate test code which combines tasks together. We normally setup a task and execute that task. But in the end it also doesn't really matter to use the register method.

Reply:
Yes, that is right. That said, if using either doesn't change things, I would prefer to go with register.